File: src/main/java/gameplay/geometry/Rectangle.java
package gameplay.geometry;

import gameplay.Point;

public class Rectangle implements Shape {
    public double x, y;
    public double width, height;

    public Rectangle(double x, double y, double width, double height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }

    public Rectangle(Point p, double width, double height) {
        this(p.x, p.y, width, height);
    }

    public int getWidth() {
        return (int) width;
    }

    public int getHeight() {
        return (int) height;
    }

    public int getX() {
        return (int) x;
    }

    public int getY() {
        return (int) y;
    }

    @Override
    public boolean contains(Point point) {
        return this.x <= point.x &&
                this.y <= point.y &&
                this.x + this.width >= point.x &&
                this.y + this.height >= point.y;
    }
} 

File: src/main/kotlin/gameplay/geometry/Rectangle.kt
package gameplay.geometry

import gameplay.Point

class Rectangle(val x: Double,
                val y: Double,
                var width: Double,
                val height: Double) : Shape {

    constructor(p: Point, width: Double, height: Double) : this(p.x, p.y, width, height)
    constructor(r: Rectangle) : this(r.x, r.y, r.width, r.height)

    val intWidth: Int
        get() = width.toInt()

    val intHeight: Int
        get() = height.toInt()

    val intX: Int
        get() = x.toInt()

    val intY: Int
        get() = y.toInt()

    override fun contains(point: Point): Boolean {
        return x <= point.x && y <= point.y && x + width >= point.x && y + height >= point.y
    }
}
